src/DynamoManipulation/MousePointManipulator.cs

```diff
@@ -192,15 +187,13 @@ protected override List<(NodeModel inputNode, double amount)> InputNodesToUpdate
 
                 // When more than one input is connected to the same slider, this
                 // method will decompose the axis corresponding to each input.
-                using (var v = GetFirstAxisComponent(item.Value.Item1))
-                {
-                    var amount = offset.Dot(v);
+                var v = GetFirstAxisComponent(item.Value.Item1);
+                var amount = offset % v;
 
-                    if (Math.Abs(amount) > MIN_OFFSET_VAL)
-                    {
-                        dynamic uiNode = item.Value.Item2;
-                        inputNodes.Add((uiNode, uiNode.Value + amount));
-                    }
+                if (Math.Abs(amount) > MIN_OFFSET_VAL)
+                {
+                    dynamic uiNode = item.Value.Item2;
```

Comment:
I know this is existing code, but do we really need dynamic here?

Reply:
I think this was `dynamic` since it was originally intended to be applicable to different types of input nodes, not just sliders. In the next line, it uses the `Value` property, which isn't defined on the `NodeModel` class, but which is the common base class for all nodes (input nodes in this case).

Reply:
so it sounds like you don't need dynamic you need to use the common base class or an interface etc?

I think using the dlr has significant performance implications.

Reply:
I actually think that @twastvedt wanted a similar interface for accessing input nodes in player without referencing the actual NodeModel types.